### PR TITLE
Completely drop CLI content-host in favour of host

### LINKED ISSUE #3939: CLI: Merge content-host and host tests and drop content-host module
```
hammer content-host --help
Error: No such sub-command 'content-host'

See: 'hammer --help'
```

Content-hosts were completely dropped from hammer (in favor of hosts). This means we need to compare content-hosts and hosts tests, add whatever is missing to hosts module and completely drop cli content-hosts module for 6.3.

### PR DESCRIPTION
**Pull request by abalakh**
Hammer's `content-host` subcommand was completely dropped in favour
of `host` in 6.3.
Replaced all the content-host calls with host calls and moved existing
content-host tests to host module (and made some fixes/improvements for
them)
Closes #3939

### CODE REVIEW

**Comment by abalakh** [comment]
Can't share import tests results as both affected tests are skipped due to BZ, end_to_end test is failing few lines before my change, host tests:
```python
py.test tests/foreman/cli/test_host.py::HostCreateTestCase -v
================================================ test session starts ================================================
platform linux2 -- Python 2.7.10, pytest-3.0.3, py-1.4.31, pluggy-0.4.0 -- /home/andrii/env/bin/python
cachedir: .cache
rootdir: /home/andrii/workspace/robottelo, inifile: 
plugins: xdist-1.13.1, repeat-0.2
collected 19 items 

tests/foreman/cli/test_host.py::HostCreateTestCase::test_negative_create_with_name PASSED
tests/foreman/cli/test_host.py::HostCreateTestCase::test_negative_create_with_unpublished_cv <- robottelo/decorators/__init__.py PASSED
tests/foreman/cli/test_host.py::HostCreateTestCase::test_negative_register_twice PASSED
tests/foreman/cli/test_host.py::HostCreateTestCase::test_positive_create_baremetal_with_bios <- robottelo/decorators/__init__.py PASSED
tests/foreman/cli/test_host.py::HostCreateTestCase::test_positive_create_baremetal_with_uefi <- robottelo/decorators/__init__.py PASSED
tests/foreman/cli/test_host.py::HostCreateTestCase::test_positive_create_using_libvirt_without_mac <- robottelo/decorators/__init__.py PASSED
tests/foreman/cli/test_host.py::HostCreateTestCase::test_positive_create_with_cv <- robottelo/decorators/__init__.py PASSED
tests/foreman/cli/test_host.py::HostCreateTestCase::test_positive_create_with_cv_default <- robottelo/decorators/__init__.py PASSED
tests/foreman/cli/test_host.py::HostCreateTestCase::test_positive_create_with_lce <- robottelo/decorators/__init__.py PASSED
tests/foreman/cli/test_host.py::HostCreateTestCase::test_positive_create_with_lce_library <- robottelo/decorators/__init__.py PASSED
tests/foreman/cli/test_host.py::HostCreateTestCase::test_positive_create_with_name PASSED
tests/foreman/cli/test_host.py::HostCreateTestCase::test_positive_create_with_org_name PASSED
tests/foreman/cli/test_host.py::HostCreateTestCase::test_positive_list PASSED
tests/foreman/cli/test_host.py::HostCreateTestCase::test_positive_register_with_no_ak PASSED
tests/foreman/cli/test_host.py::HostCreateTestCase::test_positive_unregister PASSED
tests/foreman/cli/test_host.py::HostCreateTestCase::test_positive_verify_files_with_pxegrub2_uefi <- robottelo/decorators/__init__.py PASSED
tests/foreman/cli/test_host.py::HostCreateTestCase::test_positive_verify_files_with_pxegrub2_uefi_secureboot <- robottelo/decorators/__init__.py PASSED
tests/foreman/cli/test_host.py::HostCreateTestCase::test_positive_verify_files_with_pxegrub_uefi <- robottelo/decorators/__init__.py PASSED
tests/foreman/cli/test_host.py::HostCreateTestCase::test_positive_verify_files_with_pxegrub_uefi_secureboot <- robottelo/decorators/__init__.py PASSED

============================================ 19 passed in 930.29 seconds ============================================
```

File: tests/foreman/cli/test_host.py [hunk 1]
```diff
@@ -73,6 +79,33 @@
 class HostCreateTestCase(CLITestCase):
     """Tests for creating the hosts via CLI."""
 
+    @classmethod
+    def setUpClass(cls):
+        """"""
```
Comment (sghai) [inline tests/foreman/cli/test_host.py]: would be good have one liner docstrings

File: tests/foreman/cli/test_host.py [hunk 2]
```diff
@@ -117,6 +150,248 @@ def test_positive_create_with_name(self):
...
+    @tier1
+    def test_positive_create_with_cv_default(self):
+        """Check if host can be created with content view name
+
+        @id: bb69a70e-17f9-4639-802d-90e6a4520afa
+
+        @Assert: Host is created using content view name
+        """
+        new_host = make_fake_host({
+            'content-view-id': self.DEFAULT_CV['id'],
```
Comment (sghai) [inline tests/foreman/cli/test_host.py]: As per docstrings create host w/ content-view name, however, we creating with `id` ?

File: tests/foreman/cli/test_host.py [hunk 3]
```diff
@@ -117,6 +150,248 @@ def test_positive_create_with_name(self):
...
+    def test_positive_create_with_lce_library(self):
+        """Check if host can be created with lifecycle name
+
+        @id: 0093be1c-3664-448e-87f5-758bab34958a
+
+        @Assert: Host is created using lifecycle name
+        """
+        new_host = make_fake_host({
+            'content-view-id': self.DEFAULT_CV['id'],
+            'lifecycle-environment-id': self.LIBRARY['id'],
```
Comment (sghai) [inline tests/foreman/cli/test_host.py]: lifecycle name ?

File: tests/foreman/cli/test_contenthost.py [hunk 4]
```diff
@@ -1,495 +0,0 @@
...
-                    u'lifecycle-environment-id': self.LIBRARY['id'],
-                    u'name': name,
-                    u'organization-id': self.NEW_ORG['id'],
-                })
-                ContentHost.delete({u'host-id': content_host['id']})
-                with self.assertRaises(CLIReturnCodeError):
-                    ContentHost.info({'id': content_host['id']})
-
-    @tier1
-    def test_positive_delete_by_name(self):
```
Comment (sghai) [inline tests/foreman/cli/test_contenthost.py]: its already covered under `cli/test_host.py`

**Comment by sghai** [comment]
ACK pending few comments

File: tests/foreman/cli/test_host.py (hunk 2 above)
Comment (abalakh) [inline tests/foreman/cli/test_host.py]: Wrong docstring. The test is not about creating using some cv name, but creating using default content view. I'll update the docstring

File: tests/foreman/cli/test_host.py (hunk 3 above)
Comment (abalakh) [inline tests/foreman/cli/test_host.py]: Same here

File: tests/foreman/cli/test_host.py (hunk 1 above)
Comment (abalakh) [inline tests/foreman/cli/test_host.py]: Thanks for catching. Of course, leaving the method without docstring is not an option and leaving empty doctring is not acceptable at all. That's something what i left to update later and totally forgot about it.

File: tests/foreman/cli/test_contenthost.py (hunk 4 above)
Comment (sghai) [inline tests/foreman/cli/test_contenthost.py]: nop.. thanks

**Comment by oshtaier** [comment]
There are couple of things that I cannot treat even as nitpicks =)
Checked once more code base whether we haven't missed any place for content host removal
ACK
great job!

**Comment by oshtaier** [comment]
@sghai we are ready to merge it, right?

**Comment by sghai** [comment]
@oshtaier : Yeah..  Please merge. thanks.

**Pull request merged**